### PR TITLE
Allow description override using View

### PR DESCRIPTION
**Pull request by cijothomas**
Example snippet:

```
AddView("MyFruitCounter", new MetricStreamConfiguration() { Name = "new name", Description = "modified description" })
```

### CODE REVIEW

File: src/OpenTelemetry/Metrics/MetricStreamConfiguration.cs [hunk 1]
```diff
@@ -35,8 +35,6 @@ public class MetricStreamConfiguration
 
         public string Description { get; set; }
 
-        public string Unit { get; set; }
```
Comment (cijothomas) [inline src/OpenTelemetry/Metrics/MetricStreamConfiguration.cs:38]: Unit never made it to spec...so removing.

File: src/OpenTelemetry/Metrics/Metric.cs [hunk 2]
```diff
@@ -25,15 +25,24 @@ public class Metric
...
+            string metricDescription)
+            : this(instrument, temporality, DefaultHistogramBounds, metricName, metricDescription)
         {
         }
 
-        internal Metric(Instrument instrument, AggregationTemporality temporality, double[] histogramBounds, string metricName = "")
+        internal Metric(
+            Instrument instrument,
+            AggregationTemporality temporality,
+            double[] histogramBounds,
```
Comment (utpilla) [inline src/OpenTelemetry/Metrics/Metric.cs:40]: Move `histogramBounds` to the end make its default value as `DefaultHistogramBounds`? That way we wouldn't need the other ctor?

Comment (cijothomas) [reply src/OpenTelemetry/Metrics/Metric.cs:40]: i tried that, but then realized the default must be a const, and arrays cannot be used for that purpose!

Comment (utpilla) [reply src/OpenTelemetry/Metrics/Metric.cs:40]: Have the default value of `histogramBounds` as `null` and then assign that as `histogramBounds` inside the method?

**Review by reyang — APPROVED** [review]

**Pull request merged**